### PR TITLE
TF 0.12 upgrade with CIDR issue

### PR DESCRIPTION
**Pull request by tm-g**

### CODE REVIEW

**Comment by tm-g** [comment]
When running `make create` or `make teardown` the following errors occur after upgrading to TF0.12:

Error: expected master_authorized_networks_config.0.cidr_blocks.0.cidr_block to contain a valid CIDR, got: /32 with err: invalid CIDR address: /32

  on main.tf line 84, in resource "google_container_cluster" "primary":
  84: resource "google_container_cluster" "primary" {



Error: expected master_authorized_networks_config.0.cidr_blocks.1.cidr_block to contain a valid CIDR, got: /32 with err: invalid CIDR address: /32

  on main.tf line 84, in resource "google_container_cluster" "primary":
  84: resource "google_container_cluster" "primary" {



Error: expected master_authorized_networks_config.0.cidr_blocks.2.cidr_block to contain a valid CIDR, got: /32 with err: invalid CIDR address: /32

  on main.tf line 84, in resource "google_container_cluster" "primary":
  84: resource "google_container_cluster" "primary" {


Makefile:35: recipe for target 'teardown' failed
make: *** [teardown] Error 1

**Comment by tm-g** [comment]
I believe what might be happening is the following line under `terraform/modules/instance/outputs.tf` is not returning any value for some reason:

``` terraform
output "external_ip" {
  value = google_compute_instance.instance.network_interface[0].access_config[0].ass\
igned_nat_ip
}
```

**Comment by tm-g** [comment]
test this please

**Comment by chrislovecnm** [comment]
make terraform_validate is looping on my laptop.  Also there is a single space the in Makefile.

```

diff --git a/Makefile b/Makefile
index ff146d8..18ca27b 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ ROOT := ${CURDIR}
 # lint is the first target in the file so it will get picked up when you just
 #   run 'make' on its own
 lint: check_shell check_shebangs check_python check_golang check_terraform \
-       check_docker check_base_files #check_trailing_whitespace check_headers
+       check_docker check_base_files #check_trailing_whitespace check_headers

 # create/delete/validate is for CICD
 .PHONY: create
```

**Comment by chrislovecnm** [comment]
You seem to have a bunch of files that you added??  Why??

File: terraform/modules/network/firewall/firewall.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
```
[the remark this answers is not in the document]
Comment (tm-g) [reply terraform/modules/network/firewall/firewall.tf]: I removed these in the latest push

File: terraform/modules/instance/main.tf [hunk 2]
```diff
@@ -37,74 +38,78 @@ echo "gcloud container clusters get-credentials $${cluster_name} \
...
-    cluster_name = "${var.cluster_name}"
-    zone         = "${var.zone}"
-    project      = "${var.project}"
 
-    // If this instance needs admin access, bind the user to the cluster-admin role,
-    //if it hasn't already been bound
-    admin_binding = "${var.grant_cluster_admin ?
-        "${data.null_data_source.grant_admin.outputs["command"]}" : ""}"
-  }
+vars = {
```
[the remark this answers is not in the document]
Comment (tm-g) [reply terraform/modules/instance/main.tf]: If you're referring to the `=`, that was added by running `terraform 0.12upgrade` and seems to be part of the new syntax if I'm understanding the upgrade guide correctly: https://www.terraform.io/upgrade-guides/0-12.html#attributes-vs-blocks

File: terraform/main.tf [hunk 3]
```diff
@@ -134,20 +134,20 @@ resource "google_container_cluster" "primary" {
...
-      },
-    ]
+    cidr_blocks {
+      display_name = "gke-tutorial-admin"
+      cidr_block   = join("/", [module.bastion.external_ip,"32"])
+    }
+    cidr_blocks {
+      display_name = "gke-tutorial-owner"
+      cidr_block   = join("/", [module.owner_instance.external_ip,"32"])
+//      cidr_block   = "${module.owner_instance.external_ip}/32"
```
[the remark this answers is not in the document]
Comment (tm-g) [reply terraform/main.tf]: Resolved in latest push

File: scripts/teardown.sh [hunk 4]
```diff
@@ -30,5 +30,5 @@ ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
 source "$ROOT/scripts/common.sh"
 
 # Tear down Terraform-managed resources and remove generated tfvars
-cd "$ROOT/terraform" || exit; terraform destroy -input=false -auto-approve
+cd "$ROOT/terraform" || exit; TFLOG=DEBUG terraform destroy -input=false -auto-approve
```
[the remark this answers is not in the document]
Comment (tm-g) [reply scripts/teardown.sh]: Resolved in latest push

File: terraform/modules/instance/main.tf [hunk 5]
```diff
@@ -120,13 +125,15 @@ resource "google_compute_instance" "instance" {
...
 
-  provisioner "local-exec" {
-    command = "gcloud compute scp --project ${var.project} --zone ${var.zone} --recurse ${path.module}/manifests ${var.hostname}:"
-  }
+provisioner "local-exec" {
+command = "echo \"${data.template_file.rbac_yaml.rendered}\" > '${path.module}/manifests/rbac.yaml'"
+}
+
+provisioner "local-exec" {
+command = "gcloud compute scp --project ${var.project} --zone ${var.zone} --recurse ${path.module}/manifests ${var.hostname}:"
```
[the remark this answers is not in the document]
Comment (tm-g) [reply terraform/modules/instance/main.tf]: > You seem to have a bunch of files that you added?? Why??

Unintentional. I must have fat-fingered something, and then proceeded to be clueless about it :(

**Pull request merged**